src/compiler-debugging.md

```diff
@@ -0,0 +1,320 @@
+---
+layout: default
+title: Debugging the Compiler
+---
+
+# Debugging the compiler
+[debugging]: #debugging
+
+Here are a few tips to debug the compiler:
+
+## Getting a backtrace
+[getting-a-backtrace]: #getting-a-backtrace
+
+When you have an ICE (panic in the compiler), you can set
+`RUST_BACKTRACE=1` to get the stack trace of the `panic!` like in
+normal Rust programs.  IIRC backtraces **don't work** on Mac and on MinGW,
+sorry. If you have trouble or the backtraces are full of `unknown`,
+you might want to find some way to use Linux or MSVC on Windows.
+
+In the default configuration, you don't have line numbers enabled, so the 
+backtrace looks like this:
+
+   ```
+stack backtrace:
+   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
+   1: std::sys_common::backtrace::_print
+   2: std::panicking::default_hook::{{closure}}
+   3: std::panicking::default_hook
+   4: std::panicking::rust_panic_with_hook
+   5: std::panicking::begin_panic
+   6: rustc_typeck::check::cast::<impl rustc_typeck::check::FnCtxt<'a, 'gcx, \ 
```

Comment:
Typo?

Reply:
If this line is too long, then we could just remove it altogether...

Reply:
I had to reduce the line length, so kept it in a separate line. Now removed it.